nx test: run single test instead all test in the package

### Description

Wanna ask if there are any options for the `@nx-go/nx-go:test` to run a single test. My workaround for now is:

```
go test -run TestProjection ./packages/library/backend/projection/
--- FAIL: TestProjection (0.00s)
    main_test.go:19: 
                Error Trace:    /home/rod/projects/_me/queue_port/packages/library/backend/projection/main_test.go:19
                                                        /home/rod/go/pkg/mod/go.mongodb.org/mongo-driver@v1.11.6/mongo/integration/mtest/mongotest.go:263
                Error:          Expected nil, but got: mongo.CommandError{Code:0, Message:"no responses remaining", Labels:[]string{"NetworkError", "RetryableWriteError"}, Name:"", Wrapped:no responses remaining, Raw:bson.Raw(nil)}
                Test:           TestProjection
FAIL
FAIL    github.com/rodentskie/queue_port/packages/library/backend/projection    0.004s
FAIL
```

Here is the result when I run test inside the library.

```
nx test library-backend-projection

> nx run library-backend-projection:test

Executing command: go test -v ./... -cover -race
=== RUN   TestProjection
=== RUN   TestProjection/success
=== NAME  TestProjection
    main_test.go:19: 
                Error Trace:    /home/rod/projects/_me/queue_port/packages/library/backend/projection/main_test.go:19
                                                        /home/rod/go/pkg/mod/go.mongodb.org/mongo-driver@v1.11.6/mongo/integration/mtest/mongotest.go:263
                Error:          Expected nil, but got: mongo.CommandError{Code:0, Message:"no responses remaining", Labels:[]string{"NetworkError", "RetryableWriteError"}, Name:"", Wrapped:no responses remaining, Raw:bson.Raw(nil)}
                Test:           TestProjection
--- FAIL: TestProjection (0.00s)
    --- PASS: TestProjection/success (0.00s)
FAIL
        github.com/rodentskie/queue_port/packages/library/backend/projection    coverage: 77.8% of statements
FAIL    github.com/rodentskie/queue_port/packages/library/backend/projection    0.016s
=== RUN   TestFindProjection
=== RUN   TestFindProjection/error;_not_found
=== RUN   TestFindProjection/success
--- PASS: TestFindProjection (0.00s)
    --- PASS: TestFindProjection/error;_not_found (0.00s)
    --- PASS: TestFindProjection/success (0.00s)
=== RUN   TestInsertProjection
=== RUN   TestInsertProjection/success_insert
=== RUN   TestInsertProjection/custom_error_duplicate
--- PASS: TestInsertProjection (0.00s)
    --- PASS: TestInsertProjection/success_insert (0.00s)
    --- PASS: TestInsertProjection/custom_error_duplicate (0.00s)
=== RUN   TestUpdateProjection
=== RUN   TestUpdateProjection/success
--- PASS: TestUpdateProjection (0.00s)
    --- PASS: TestUpdateProjection/success (0.00s)
PASS
        github.com/rodentskie/queue_port/packages/library/backend/projection/helper     coverage: 89.5% of statements
ok      github.com/rodentskie/queue_port/packages/library/backend/projection/helper     (cached)        coverage: 89.5% of statements
FAIL
Failed to execute command: go test -v ./... -cover -race Error: Command failed: go test -v ./... -cover -race
    at checkExecSyncError (node:child_process:861:11)
    at execSync (node:child_process:932:15)
    at runGoCommand (/home/rod/projects/_me/queue_port/node_modules/@nx-go/nx-go/src/utils/run-go-command.helper.js:13:38)
    at /home/rod/projects/_me/queue_port/node_modules/@nx-go/nx-go/src/executors/test/executor.js:14:41
    at Generator.next (<anonymous>)
    at /home/rod/projects/_me/queue_port/node_modules/tslib/tslib.js:167:75
    at new Promise (<anonymous>)
    at __awaiter (/home/rod/projects/_me/queue_port/node_modules/tslib/tslib.js:163:16)
    at runExecutor (/home/rod/projects/_me/queue_port/node_modules/@nx-go/nx-go/src/executors/test/executor.js:7:34) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 28769,
  stdout: null,
  stderr: null
}

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target test for project library-backend-projection (840ms)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```

--- a/docs/executors/test.md
+++ b/docs/executors/test.md
@@ -16,6 +16,10 @@ Uses `go test` command to run tests of a Go project.
 
 - (boolean): Enable race detector
 
+### run
+
+- (string): Run only tests matching this regular expression
+
 ### verbose
 
 - (boolean): Enable verbose test output

--- a/packages/nx-go-e2e/shared/create-test-project.ts
+++ b/packages/nx-go-e2e/shared/create-test-project.ts
@@ -1,14 +1,15 @@
 import { joinPathFragments, readJsonFile, workspaceRoot } from '@nx/devkit';
+import { tmpProjPath } from '@nx/plugin/testing';
 import { execSync } from 'child_process';
 import { mkdirSync, rmSync } from 'fs';
-import { dirname, join } from 'path';
+import { dirname } from 'path';
 
 /**
  * Creates a test project with create-nx-workspace and installs the plugin
  */
-export default function createTestProject(preset = 'apps') {
+export default function createTestProject(preset = 'apps'): string {
   const projectName = 'proj';
-  const projectDirectory = join(process.cwd(), 'tmp', 'nx-e2e', projectName);
+  const projectDirectory = tmpProjPath();
 
   // Ensure projectDirectory is empty
   rmSync(projectDirectory, { recursive: true, force: true });

--- a/packages/nx-go-e2e/src/nx-go.spec.ts
+++ b/packages/nx-go-e2e/src/nx-go.spec.ts
@@ -2,7 +2,7 @@ import {
   checkFilesExist,
   readFile,
   readJson,
-  runCommandAsync,
+  runNxCommandAsync,
   uniq,
   updateFile,
 } from '@nx/plugin/testing';
@@ -32,10 +32,6 @@ describe('nx-go', () => {
   afterAll(() => {
     rmSync(projectDirectory, { recursive: true, force: true });
   });
-
-  async function runNxCommandAsync(command: string) {
-    return runCommandAsync(`nx ${command}`, { cwd: projectDirectory });
-  }
 
   it('should be installed', () => {
     // npm ls will fail if the package is not installed properly

--- a/packages/nx-go/src/executors/test/executor.spec.ts
+++ b/packages/nx-go/src/executors/test/executor.spec.ts
@@ -28,17 +28,17 @@ describe('Test Executor', () => {
 
   it.each`
     config                              | flag
-    ${{ verbose: true }}                | ${'-verbose'}
+    ${{ verbose: true }}                | ${'-v'}
     ${{ cover: true }}                  | ${'-cover'}
-    ${{ coverProfile: 'coverage.out' }} | ${'-coverageprofile=coverage.out'}
+    ${{ coverProfile: 'coverage.out' }} | ${'-coverprofile=coverage.out'}
     ${{ race: true }}                   | ${'-race'}
+    ${{ run: 'TestProjection' }}        | ${'-run=TestProjection'}
   `('should add flag $flag if enabled', async ({ config, flag }) => {
     const spyExecute = jest.spyOn(commonFunctions, 'executeCommand');
     const output = await executor({ ...options, ...config }, context);
     expect(output.success).toBeTruthy();
-    expect(spyExecute).toHaveBeenCalledWith(
-      expect.not.arrayContaining([flag]),
-      { cwd: 'apps/project' }
-    );
+    expect(spyExecute).toHaveBeenCalledWith(expect.arrayContaining([flag]), {
+      cwd: 'apps/project',
+    });
   });
 });

--- a/packages/nx-go/src/executors/test/executor.ts
+++ b/packages/nx-go/src/executors/test/executor.ts
@@ -19,6 +19,7 @@ export default async function runExecutor(
       ...buildFlagIfEnabled('-cover', options.cover),
       ...buildStringFlagIfValid(`-coverprofile`, options.coverProfile),
       ...buildFlagIfEnabled('-race', options.race),
+      ...buildStringFlagIfValid(`-run`, options.run),
       './...',
     ],
     { cwd: extractProjectRoot(context) }

--- a/packages/nx-go/src/executors/test/schema.d.ts
+++ b/packages/nx-go/src/executors/test/schema.d.ts
@@ -2,5 +2,6 @@ export interface TestExecutorSchema {
   cover?: boolean;
   coverProfile?: string;
   race?: boolean;
+  run?: string;
   verbose?: boolean;
 }

--- a/packages/nx-go/src/executors/test/schema.json
+++ b/packages/nx-go/src/executors/test/schema.json
@@ -20,6 +20,10 @@
       "type": "boolean",
       "default": false
     },
+    "run": {
+      "description": "Run only tests matching this regular expression",
+      "type": "string"
+    },
     "verbose": {
       "description": "Whether to enable verbose test output",
       "type": "boolean",

